### PR TITLE
[Asgardeo docs] Add Action Management Rest API docs 

### PR DESCRIPTION
**Pull request by ashanthamara**
## Related Issue:
- https://github.com/wso2-enterprise/asgardeo-product/issues/25503

### CODE REVIEW

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '500':
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
```
Comment (AnuradhaSK) [inline en/asgardeo/docs/apis/restapis/actions.yaml:39]: example curls are missing for each endpoint

Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml:44]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/69f8f822c61bc848415056485d62bc91980b7bc0

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 2]
```diff
@@ -0,0 +1,859 @@
+openapi: 3.0.0
+info:
+  version: v1
+  title: Asgardeo - Action Management Rest API
+  description:
+    "This is the RESTful API for managing action configuration for an organization in WSO2 Asgardeo. \n
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:6]: why do we say "WSO2 Asgardeo" ? Let's just say as "Asgardeo"

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 3]
```diff
@@ -0,0 +1,859 @@
...
+    This API allows creating, retrieving, updating and deleting the action configurations. \n\n
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:7]: API allows to create, retrieve, update and delete actions

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 4]
```diff
@@ -0,0 +1,859 @@
...
+        - name: actionType
+          in: path
+          description: Name of the Action Type.
+          required: true
+          schema:
+            enum:
+              - preIssueAccessToken
+              - preUpdatePassword
+              - preUpdateProfile
+              - preRegistration
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:63]: Better not have unsupported types as of now

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 5]
```diff
@@ -0,0 +1,859 @@
...
+          source: |
+            curl --location 'https://api.asgardeo.io/t/{organization-name}/api/server/v1/actions/{actionType}' \
+            -H 'Authorization: Bearer {bearer_token}'
+
+  /actions/{actionType}/{actionId}:
+    patch:
+      tags:
+        - Actions Endpoint
+      summary: Update Action
+      description: "This API updates action configurations and return the updated action configurations. \n\n
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:168]: 'action configuration' and 'action', both are used to explain what's managed with this API. Use a consistent term. And IMO, it should be 'actions'. And also this API should link back to the document that explains actions once that's available. Let's keep a note of it

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 6]
```diff
@@ -0,0 +1,859 @@
...
+          required: true
+          schema:
+            enum:
+              - preIssueAccessToken
+              - preUpdatePassword
+              - preUpdateProfile
+              - preRegistration
+        - name: actionId
+          in: path
+          description: Id of the Action.
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:184]: Unique identifier of the action

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 7]
```diff
@@ -0,0 +1,859 @@
...
+          source: |
+            curl --location --request POST 'https://api.asgardeo.io/t/{organization-name}/api/server/v1/actions/{actionType}/{actionId}/deactivate' \
+            -H 'Authorization: Bearer {bearer_token}'
+
+  /actions/{actionType}/{actionId}/{authType}:
+    put:
+      tags:
+        - Actions Endpoint
+      summary: Update Action Authentication
+      description: "This API updates action endpoint authentication and return the updated action configurations. \n\n
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:428]: Updates authentication mechanism for the endpoint configured for the action

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 8]
```diff
@@ -0,0 +1,859 @@
...
+              - none
+              - basic
+              - apiKey
+              - bearer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AuthTypeProperties'
+        description: This represents the action endpoint authentication to be updated.
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:463]: Fix based on comment above

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 9]
```diff
@@ -0,0 +1,859 @@
...
+        type:
+          type: string
+          enum:
+            - PRE_ISSUE_ACCESS_TOKEN
+            - PRE_UPDATE_PASSWORD
+            - PRE_UPDATE_PROFILE
+            - PRE_REGISTRATION
+        name:
+          type: string
+          example: Access Token Pre Issue
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:615]: This example should be using the defined enum isn't it ?

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 10]
```diff
@@ -0,0 +1,859 @@
...
+          minLength: 1
+          maxLength: 255
+        description:
+          type: string
+          example: This is the configuration of pre-action for issuing access token.
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:620]: This action invokes before issuing an access token

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 11]
```diff
@@ -0,0 +1,859 @@
...
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:650]: Fix based on comment above

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 12]
```diff
@@ -0,0 +1,859 @@
...
+          example: "Pre Issue Access Token."
+        description:
+          type: string
+          example: "Extension point configuration for Pre Issue Access Token."
+        count:
+          type: integer
+          example: 2
+        self:
+          type: string
+          example: "/t/{organization-name}/api/server/v1/actions/preIssueAccessToken"
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:689]: Ideally each property should have a description explaning what this property is about. In minimal, count should have

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 13]
```diff
@@ -0,0 +1,859 @@
...
+          type: string
+          example: Access Token Pre Issue
+          minLength: 1
+          maxLength: 255
+        description:
+          type: string
+          example: This is the configuration of pre-action for issuing access token.
+          maxLength: 255
+        endpoint:
+          $ref: '#/components/schemas/EndpointUpdateModel'
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:718]: Ideally these properties should have a description

File: en/asgardeo/docs/apis/restapis/actions.yaml [hunk 14]
```diff
@@ -0,0 +1,859 @@
...
+
+    EndpointUpdateModel:
+      type: object
+      properties:
+        uri:
+          type: string
+          example: https://myextension.com/token
+          pattern: '^https?://.+'
+        authentication:
+          $ref: '#/components/schemas/Authentication'
```
Comment (malithie) [inline en/asgardeo/docs/apis/restapis/actions.yaml:728]: These properties should have a description

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 9 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml:615]: This is the action name. Only the action type comes with the enum

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 2 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 4 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 3 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 5 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 7 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 11 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 12 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 13 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 14 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 6 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

File: en/asgardeo/docs/apis/restapis/actions.yaml (hunk 8 above)
Comment (ashanthamara) [reply en/asgardeo/docs/apis/restapis/actions.yaml]: Fixed with https://github.com/wso2/docs-is/pull/4698/commits/ee4f5ca7c24d3d8dcaca4943b7b5aadb6e16e7fc

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**